src/subdomains/core/buy-crypto/process/services/buy-crypto.service.ts

```diff
@@ -896,6 +899,18 @@ export class BuyCryptoService {
     return { volume: volume ?? 0, credit: credit ?? 0 };
   }
 
+  async getPartnerFeeRefVolume(ref: string): Promise<{ volume: number; credit: number }> {
+    const { volume, credit } = await this.buyCryptoRepo
+      .createQueryBuilder('buyCrypto')
+      .select('SUM((partnerFeeAmount * (amountInEur/amountInChf )) / (refProvision * 0.01))', 'volume')
+      .addSelect('SUM(partnerFeeAmount * (amountInEur/amountInChf ))', 'credit')
```

Comment:
Isn't the volume just `amountInEur`?

Reply:
Nope the volume is always calculated with the refProvision of the user, normally we would take the amountInEur and calculate the credit with it but in this scenario the credit is given with the partnerFeeAmount and we have to calculate the right volume for it to match the ref calculation